packages/diagrams/frontend/sirius-components-diagrams/src/renderer/actions/visibility/hooks/useManageVisibilityInvokeAction.tsx

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { gql, useMutation } from '@apollo/client';
+import { useMultiToast } from '@eclipse-sirius/sirius-components-core';
+import { useContext, useEffect } from 'react';
+import { DiagramContext } from '../../../../contexts/DiagramContext';
+import { DiagramContextValue } from '../../../../contexts/DiagramContext.types';
+import { GQLManageVisibilityAction } from './useManageVisibilityActions.types';
+import {
+  GQLErrorPayload,
+  GQLManageVisibilityInvokeActionData,
+  GQLManageVisibilityInvokeActionInput,
+  GQLManageVisibilityInvokeActionPayload,
+  GQLManageVisibilityInvokeActionVariables,
+  GQLSuccessPayload,
+  UseManageVisibilityInvokeActionValue,
+} from './useManageVisibilityInvokeAction.types';
+
+const invokeActionMutation = gql`
+  mutation invokeManageVisibilityAction($input: InvokeManageVisibilityActionInput!) {
+    invokeManageVisibilityAction(input: $input) {
+      __typename
+      ... on SuccessPayload {
+        messages {
+          body
+          level
+        }
+      }
+      ... on ErrorPayload {
+        messages {
+          body
+          level
+        }
+      }
+    }
+  }
+`;
+
+const isErrorPayload = (payload: GQLManageVisibilityInvokeActionPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+const isSuccessPayload = (payload: GQLManageVisibilityInvokeActionPayload): payload is GQLSuccessPayload =>
+  payload.__typename === 'SuccessPayload';
+
+export const useManageVisibilityInvokeAction = (diagramElementId: string): UseManageVisibilityInvokeActionValue => {
```

Comment:
ask for this kind of parameter in the function returned because it can create some issue in case of the update of the props

Reply:
Done